packages/babel-preset-env/src/index.js

```diff
@@ -230,12 +230,24 @@ export default declare((api, opts) => {
   }
 
   const plugins = [];
+  const overrides = [];
   const pluginUseBuiltIns = useBuiltIns !== false;
 
-  // NOTE: not giving spec here yet to avoid compatibility issues when
-  // transform-modules-commonjs gets its spec mode
   if (modules !== false && moduleTransformations[modules]) {
-    plugins.push([getPlugin(moduleTransformations[modules]), { loose }]);
+    // NOTE: not giving spec here yet to avoid compatibility issues when
+    // transform-modules-commonjs gets its spec mode
+    const item = [getPlugin(moduleTransformations[modules]), { loose }];
+
+    if (modules === "auto") {
+      // When the `exclude` callback returns a truthy value, Babel not use the
+      // 'item' plugin here.
+      overrides.push({
+        exclude: (filename, { caller }) => caller && caller.supportsStaticESM,
```

Comment:
Since the caller isn't file-specific, can it be passed via the `api` object instead of using the `exclude` option, which is usually used to exclude plugins based on the file path?